enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/integrations/kernel/FulltextIndexProvider.java

```diff
@@ -0,0 +1,230 @@
+        }
+        catch ( IOException e )
+        {
+            return InternalIndexState.POPULATING;
+        }
+    }
+
+    private boolean indexIsOnline( long indexId, FulltextIndexDescriptor descriptor ) throws IOException
+    {
+        try ( LuceneFulltext index = factory.createFulltextIndex( indexId, descriptor ) )
+        {
+            if ( index.exists() )
+            {
+                index.open();
+                return index.isOnline();
+            }
+            return false;
+        }
+    }
+
+    @Override
+    public IndexCapability getCapability( FulltextIndexDescriptor indexDescriptor )
+    {
+        return IndexCapability.NO_CAPABILITY;
+    }
+
+    @Override
+    public boolean compatible( IndexDescriptor indexDescriptor )
+    {
+        return indexDescriptor instanceof FulltextIndexDescriptor;
+    }
+
+    @Override
+    public StoreMigrationParticipant storeMigrationParticipant( FileSystemAbstraction fs, PageCache pageCache )
+    {
+        return new SchemaIndexMigrator( fs, this );
+    }
+
+    @Override
+    public Stream<String> propertyKeyStrings( IndexDescriptor descriptor )
+    {
+        return Arrays.stream( descriptor.schema().getPropertyIds() ).mapToObj( id -> propertyKeyTokenHolder.getTokenByIdOrNull( id ).name() );
+    }
+
+    @Override
+    public IndexDescriptor indexDescriptorFor( String name, EntityType type, String[] entityTokens, String... properties )
+    {
+        if ( Arrays.stream( properties ).anyMatch( prop -> prop.equals( FulltextAccessor.FIELD_ENTITY_ID ) ) )
+        {
+            throw new BadSchemaException( "Unable to index the property " + FulltextAccessor.FIELD_ENTITY_ID );
+        }
+        int[] entityTokenIds;
+        if ( type == EntityType.NODE )
+        {
+            entityTokenIds = Arrays.stream( entityTokens ).mapToInt( labelTokenHolder::getOrCreateId ).toArray();
+        }
+        else
+        {
+            entityTokenIds = Arrays.stream( entityTokens ).mapToInt( relationshipTypeTokenHolder::getOrCreateId ).toArray();
+        }
+        int[] propertyIds = Arrays.stream( properties ).mapToInt( propertyKeyTokenHolder::getOrCreateId ).toArray();
+        return indexDescriptorFor( SchemaDescriptorFactory.multiToken( entityTokenIds, type, propertyIds ), name, analyzerClassName );
+    }
+
+    @Override
+    public ScoreEntityIterator query( String indexName, String queryString ) throws IOException, IndexNotFoundKernelException
+    {
+        FulltextIndexAccessor fulltextIndexAccessor = accessorsByName.get( indexName );
+        if ( fulltextIndexAccessor == null )
+        {
+            throw new IndexNotFoundKernelException( "The requested fulltext index could not be accessed. Perhaps population has not completed yet?" );
+        }
+        return fulltextIndexAccessor.query( queryString );
+    }
+
+    private class BadSchemaException extends IllegalArgumentException
+    {
+        BadSchemaException( String message )
```

Comment:
bad schema error? with the only usage with a message that its impossible to index property?